src/genome.c

```diff
@@ -20,6 +20,42 @@
 #include "scalam.h"
 
 /**
+ * @brief Creates a single upgrade step consisting of a set of programs,
+ *        their versions/commits and whether they are installed or not
+ * @param population The population in which the genome exists
+ * @param individual The genome to be mutated
+ * @param upgrade_step The index within the upgrade series
+ * @returns zero on success
+ */
+int genome_create_installation_step(sc_population * population,
+									sc_genome * individual,
+									int upgrade_step)
+{
+	int prog_index;
+
+	/* for each possible program within the system */
+	for (prog_index = 0;
+		 prog_index < population->sys.no_of_programs;
+		 prog_index++) {
+
+		/* check that there are some versions/commits for htis program */
```

Comment:
/* check that there are some versions/commits for htis program */
to
/* check that there are some versions/commits for this program */